lib/__tests__/standalone-fix.test.mjs

```diff
@@ -402,3 +402,67 @@ it('returns the original code if the fix option is true but the code is not fixa
 		× 1 problem (1 error, 0 warnings)
 	`);
 });
+
+it("doesn't fix when rules always attempt to apply fixes", async () => {
+	const code = `
+		.foo {}`;
+
+	const result = await standalone({
+		code,
+		config: {
+			plugins: [fixturesPath('./plugin-fixes-always.mjs')],
+			rules: {
+				'plugin/fixes-always': true,
+			},
+		},
+		fix: false,
+		formatter: 'string',
+	});
+
+	expect(result.code).toBeUndefined();
+	expect(result.report).toBe('');
+});
+
+it("doesn't fix with stylelint-disable commands when rules always attempt to apply fixes", async () => {
+	const code = `
+		/* stylelint-disable */
+		.foo {}`;
+
+	const result = await standalone({
+		code,
+		config: {
+			plugins: [fixturesPath('./plugin-fixes-always.mjs')],
+			rules: {
+				'plugin/fixes-always': true,
+			},
+		},
+		fix: true,
+		formatter: 'string',
+	});
+
+	expect(result.code).toBe(code);
+	expect(result.report).toBe('');
+});
+
+it('does fix with scoped stylelint-disable commands when rules always attempt to apply fixes', async () => {
+	const code = stripIndent`
+		/* stylelint-disable plugin/fixes-always */
+		.foo {}`;
+
+	const result = await standalone({
+		code,
+		config: {
+			plugins: [fixturesPath('./plugin-fixes-always.mjs')],
+			rules: {
+				'plugin/fixes-always': true,
+			},
+		},
+		fix: true,
+		formatter: 'string',
+	});
+
+	expect(result.code).toBe(stripIndent`
+		/* stylelint-disable plugin/fixes-always */
+		.bar {}`);
+	expect(result.report).toBe('');
+});
```

Comment:
Notice how rule specific disable comments also do not work today.

`postcssResult.stylelint.disableWritingFix` does not prevent writes in this scenario.

Reply:
Indeed the order of the booleans* should have been different to cater for the plugins not respecting the convention of using `context.fix` block for fixes.

*l114-118 lintPostcssResult